src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,804 @@
+                // For first part we keep slashes, like '/top', 'C:\' or 'phar://'
+                $finalPath = $path;
+                $wasScheme = (false !== strpos($path, '://'));
+                continue;
+            }
+
+            // Only add slash if previous part didn't end with '/' or '\'
+            if (!\in_array(substr($finalPath, -1), ['/', '\\'])) {
+                $finalPath .= '/';
+            }
+
+            // If first part included a scheme like 'phar://' we allow current part to start with '/', otherwise trim
+            $finalPath .= $wasScheme ? $path : ltrim($path, '/');
+            $wasScheme = false;
+        }
+
+        if (null === $finalPath) {
+            return '';
+        }
+
+        return self::canonicalize($finalPath);
+    }
+
+    /**
+     * Returns whether a path is a base path of another path.
+     *
+     * Dot segments ("." and "..") are removed/collapsed and all slashes turned
+     * into forward slashes.
+     *
+     * ```php
+     * Path::isBasePath('/webmozart', '/webmozart/css');
+     * // => true
+     *
+     * Path::isBasePath('/webmozart', '/webmozart');
+     * // => true
+     *
+     * Path::isBasePath('/webmozart', '/webmozart/..');
+     * // => false
+     *
+     * Path::isBasePath('/webmozart', '/puli');
+     * // => false
+     * ```
+     */
+    public static function isBasePath(string $basePath, string $ofPath): bool
+    {
+        $basePath = self::canonicalize($basePath);
+        $ofPath = self::canonicalize($ofPath);
+
+        // Append slashes to prevent false positives when two paths have
+        // a common prefix, for example /base/foo and /base/foobar.
+        // Don't append a slash for the root "/", because then that root
+        // won't be discovered as common prefix ("//" is not a prefix of
+        // "/foobar/").
+        return 0 === strpos($ofPath.'/', rtrim($basePath, '/').'/');
+    }
+
+    /**
+     * Splits a canonical path into its root directory and the remainder.
+     *
+     * If the path has no root directory, an empty root directory will be
+     * returned.
+     *
+     * If the root directory is a Windows style partition, the resulting root
+     * will always contain a trailing slash.
+     *
+     * list ($root, $path) = Path::split("C:/webmozart")
+     * // => ["C:/", "webmozart"]
+     *
+     * list ($root, $path) = Path::split("C:")
+     * // => ["C:/", ""]
+     *
+     * @return string[] an array with the root directory and the remaining
+     *                  relative path
+     */
+    private static function split(string $path): array
+    {
+        if ('' === $path) {
+            return ['', ''];
+        }
+
+        // Remember scheme as part of the root, if any
+        if (false !== ($pos = strpos($path, '://'))) {
+            $root = substr($path, 0, $pos + 3);
+            $path = substr($path, $pos + 3);
+        } else {
+            $root = '';
+        }
+
+        $length = \strlen($path);
```

Comment:
Should `mb_strlen` be used when possible? Same question for all `substr`,`strpos`... added here

Reply:
Hm might go for a round of extra tests for it if there is not already